src/query.rs

```diff
@@ -114,6 +146,16 @@ Failed to parse a query: {e}
 
         queries
     }
+
+    pub fn apply_override(&mut self, query_override: &QueryOverride) {
+        if let Some(lint_level) = query_override.lint_level {
+            self.lint_level = lint_level;
+        }
+
+        if let Some(required_update) = query_override.required_update {
+            self.required_update = required_update;
+        }
+    }
```

Comment:
A related piece of work-in-progress that might be of interest is this draft PR: https://github.com/obi1kenobi/cargo-semver-checks/pull/600/files

It proposes a design for describing "the plan" for what to do if various kinds of problems arise while linting. In the future, we'll want both the "what things run on which crates" plan I described earlier, and the "what if something goes wrong" bit here to be part of the same inspectable, testable plan.

Reply:
Sorry for the late response - school got a little crazy.  

Thanks for the feedback! It definitely makes sense that when checking a crate, we want to have pass in the effective levels of each lint, both for semver requirement and lint level.  We can calculate the effective lint level for each crate to be checked and pass it as an argument (mapping from lint name to `{ required_update, lint_level }`) to `check_release` (and add it to the `CrateCheck` struct in the proposed API).  We could also just pass a map that only has an entry for a given lint if it differs from the default value in the lint file.

Passing *all* effective values might be easier to use, and to test in the proposed API version, as well as potentially easier to calculate multiple levels of overrides, but passing only ones that are different from the default would save space and execution time, but it might be a little more complicated and potentially susceptible to bugs about calculating and using the right effective version.

Which of those seems like the better idea? Or perhaps something else? Thanks!

Reply:
No worries at all, as you know I've been quite busy too :)

From a testing perspective, I think it would be easier to pass only the overrides ("different from default"). That way, we won't have to regenerate the snapshot test data for every newly-added lint, like we would have to if all computed values were part of the passed data.

Reply:
Okay, sounds good! Thanks!